pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -107,19 +107,9 @@ func (p *LaunchTemplateProvider) Get(ctx context.Context, constraints *v1alpha1.
 	// Construct launch templates
 	launchTemplates := map[string][]cloudprovider.InstanceType{}
 	for amiID, instanceTypes := range amis {
-		// Get userData for Node
-		var userData string
-		if constraints.AMIFamily != nil {
-			if strings.EqualFold(*constraints.AMIFamily, v1alpha1.OperatingSystemBottleRocket) {
-				userData, err = p.getBottleRocketUserData(ctx, constraints, additionalLabels)
-			} else if strings.EqualFold(*constraints.AMIFamily, v1alpha1.OperatingSystemEKSOptimized) {
-				userData, err = p.getEKSOptimizedUserData(ctx, constraints, instanceTypes, additionalLabels)
-			} else {
-				logging.FromContext(ctx).Warnf("AMIFamily was set, but was not one of %s or %s. Setting to %s as the default.", v1alpha1.OperatingSystemEKSOptimized, v1alpha1.OperatingSystemBottleRocket, v1alpha1.OperatingSystemEKSOptimized)
-				userData, err = p.getEKSOptimizedUserData(ctx, constraints, instanceTypes, additionalLabels)
-			}
-		} else {
-			userData, err = p.getEKSOptimizedUserData(ctx, constraints, instanceTypes, additionalLabels)
+		userData, err := p.getEKSOptimizedUserData(ctx, constraints, instanceTypes, additionalLabels)
```

Comment:
Can we get and pass in cabundle before to simplify this error handling?

Reply:
yep